handlerconfigurer.go

```diff
@@ -0,0 +1,191 @@
+package configkit
+
+import (
+	"strings"
+
+	"github.com/dogmatiq/configkit/message"
+	"github.com/dogmatiq/dogma"
+)
+
+// handlerConfigurer is an implementation of the the configurer interfaces for
+// all of the Dogma handler types.
+//
+// - dogma.AggregateConfigurer
+// - dogma.ProcessConfigurer
+// - dogma.IntegrationConfigurer
+// - dogma.ProjectionConfigurer
+type handlerConfigurer struct {
+	// interfaceName is the name of the Dogma interface that the configurer is being used as.
+	// It is used to construct error messages.
+	interfaceName string
+
+	// target is the handler to populate with the configuration values.
+	target *handler
+}
+
+func (c *handlerConfigurer) Identity(name string, key string) {
+	if !c.target.ident.IsZero() {
+		Panicf(
+			"%s is configured with multiple identities (%s and %s/%s), Identity() must be called exactly once within Configure()",
+			c.target.rt.String(),
+			c.target.ident,
+			name,
+			key,
+		)
+	}
+
+	var err error
+	c.target.ident, err = NewIdentity(name, key)
+
+	if err != nil {
+		Panicf(
+			"%s is configured with an invalid identity, %s",
+			c.target.rt.String(),
+			err,
+		)
+	}
+}
+
+func (c *handlerConfigurer) ConsumesCommandType(m dogma.Message) {
+	c.consumes(m, message.CommandRole)
+}
+
+func (c *handlerConfigurer) ConsumesEventType(m dogma.Message) {
+	c.consumes(m, message.EventRole)
+}
+
+func (c *handlerConfigurer) ProducesCommandType(m dogma.Message) {
+	c.produces(m, message.CommandRole)
+}
+
+func (c *handlerConfigurer) ProducesEventType(m dogma.Message) {
+	c.produces(m, message.EventRole)
+}
+
+func (c *handlerConfigurer) SchedulesTimeoutType(m dogma.Message) {
+	c.consumes(m, message.TimeoutRole)
+	c.produces(m, message.TimeoutRole)
+}
+
+func (c *handlerConfigurer) consumes(m dogma.Message, r message.Role) {
+	mt := message.TypeOf(m)
+	c.guardAgainstRoleMismatch(mt, r)
+
+	if c.target.types.Consumed.Has(mt) {
+		verb := "consume"
+		if r == message.TimeoutRole {
+			verb = "schedule"
```

Comment:
Timeouts can be consumed also? The `SchedulesTimeoutType()` method above has timeouts for both `consumes()` and `produces()`

Reply:
Maybe this `"schedule"` thing should be in the `produces()` method below?

Reply:
In the case of this kind of failure it never reaches the `produces()` method because `consumes()` is called first in `SchedulesTimeoutMessage()`.

Reply:
But that "schedule" doesn't make sense there to me.

Reply:
A timeout is always consumed so changing the message here isn't needed? Its the procdues where that verb should be changed to be "schedule"?

Everything else is fine, I will approve the PR now. Its just that special case for timeouts in consume vs produce methods that confuses me.

Reply:
It's to make the wording of [the error message](https://github.com/dogmatiq/configkit/blob/98c2b9a0aef0b7bacc8628f2dd9623aac54ac7a0/process_test.go#L247) match the wording of [the public interface](https://github.com/dogmatiq/configkit/blob/98c2b9a0aef0b7bacc8628f2dd9623aac54ac7a0/handlerconfigurer.go#L65).